src/test/java/uk/ac/ebi/subs/biostudies/client/BioStudiesClientTest.java

```diff
@@ -0,0 +1,69 @@
+package uk.ac.ebi.subs.biostudies.client;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import uk.ac.ebi.subs.biostudies.BioStudiesAgentApp;
+import uk.ac.ebi.subs.biostudies.TestUtil;
+import uk.ac.ebi.subs.biostudies.model.BioStudiesSubmission;
+
+import java.util.UUID;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringBootTest(classes = BioStudiesAgentApp.class)
+public class BioStudiesClientTest {
+
+    @Autowired
+    private BioStudiesConfig config;
+    private BioStudiesSubmission bioStudiesSubmission;
+
+    @Before
+    public void buildup() {
+
+        bioStudiesSubmission = (BioStudiesSubmission) TestUtil.loadObjectFromJson(
+                "exampleProject_biostudies.json", BioStudiesSubmission.class
+        );
+    }
+
+    @Test
+    public void login() {
+        BioStudiesClient client = new BioStudiesClient(config);
+        BioStudiesSession session = client.initialiseSession();
+
+        Assert.assertEquals("OK", session.getBioStudiesLoginResponse().getStatus());
+        Assert.assertNotNull(session.getBioStudiesLoginResponse().getSessid());
+        System.out.println(session);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
```

Comment:
This is fine, but if you also want to check the message of the exception, then you can use the `@Rule` based exception checking.
`@Rule
  public ExpectedException exception = ExpectedException.none();`

then in the method:
`exception.expect(IllegalArgumentException.class);
  exception.expectMessage(containsString('...'));`